test/EndToEndTests/Tests/Client/Microsoft.OData.Client.E2E.Tests/ClientTests/Server/OpenTypesServiceClientTestsController.cs

```diff
@@ -0,0 +1,60 @@
+﻿//-----------------------------------------------------------------------------
+// <copyright file="OpenTypesServiceClientTestsController.cs" company=".NET Foundation">
+//      Copyright (c) .NET Foundation and Contributors. All rights reserved.
+//      See License.txt in the project root for license information.
+// </copyright>
+//------------------------------------------------------------------------------
+
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.OData.Formatter;
+using Microsoft.AspNetCore.OData.Query;
+using Microsoft.AspNetCore.OData.Routing.Controllers;
+using Microsoft.OData.Client.E2E.Tests.Common.Server.OpenTypes;
+
+namespace Microsoft.OData.Client.E2E.Tests.ClientTests.Server
+{
+    public class OpenTypesServiceClientTestsController : ODataController
+    {
+        [EnableQuery]
+        [HttpGet("odata/Row")]
+        public IActionResult Get()
+        {
+            var row = OpenTypesServiceDataSource.Row;

```

Comment:
Is `Row` is a collection property? If it is, it's quite intuitive to name it `Row`. `Rows` would be more intuitive and appropriate. In addition, the route template would be `odata/Rows` since the endpoint returns a collection